plugins/s3/src/index.mjs

```diff
@@ -1106,6 +1186,265 @@ const GetObject = {
   },
 }
 
+const GetObjectAcl = {
+  awsDoc: docRoot + 'API_GetObjectAcl.html',
+  validate: {
+    Bucket,
+    Key,
+    VersionId,
+    ...getValidateHeaders('ExpectedBucketOwner', 'RequestPayer'),
+  },
+  request: (params, utils) => {
+    const { Key } = params
+    const queryParams = [ 'VersionId' ]
+    const headers = getHeadersFromParams(params, queryParams)
+    const query = { 'acl': '', ...getQueryFromParams(params, queryParams) }
+    const { host, pathPrefix } = getHost(params, utils)
+    return {
+      host,
+      pathPrefix,
+      path: `/${Key}`,
+      headers,
+      query,
+    }
+  },
+  response: ({ payload, headers }) => {
+    const { AccessControlList, Owner } = payload
+    let { Grant: Grants } = AccessControlList
+    if (!Array.isArray(Grants)) {
+      Grants = [ Grants ]
+    }
+    Grants = Grants.map(i => {
+      let result = { ...i }
```

Comment:
`i` can just be `result` without a spread, we're already mutating the referenced object here.